crates/go/src/wit_runtime.go

```diff
@@ -0,0 +1,83 @@
+package wit_runtime
+
+import (
+	"fmt"
+	"runtime"
+	"unsafe"
+)
+
+func Allocate(pinner *runtime.Pinner, size, align uintptr) unsafe.Pointer {
+	pointer := allocateRaw(size, align)
+	pinner.Pin(pointer)
+	return pointer
+}
+
+func ceiling(n, d uintptr) uintptr {
+	var hasRemainder uintptr
+	if n%d == 0 {
+		hasRemainder = 0
+	} else {
+		hasRemainder = 1
+	}
+	return n/d + hasRemainder
+}
+
+func allocateRaw(size, align uintptr) unsafe.Pointer {
+	if size == 0 {
+		return unsafe.Pointer(uintptr(0))
+	}
+
+	switch align {
+	case 1:
+		return unsafe.Pointer(unsafe.SliceData(make([]uint8, size)))
+	case 2:
+		return unsafe.Pointer(unsafe.SliceData(make([]uint16, ceiling(size, align))))
+	case 4:
+		return unsafe.Pointer(unsafe.SliceData(make([]uint32, ceiling(size, align))))
+	case 8:
+		return unsafe.Pointer(unsafe.SliceData(make([]uint64, ceiling(size, align))))
```

Comment:
Did you find the calls to `ceiling` necessary or were those copy/pasted from elsewhere? I would otherwise expect them to not be necessary as the canonical ABI should guarantee that `size % align == 0` so `size / align` would be an appropriate calculation here